smarts/core/lanepoints.py

```diff
@@ -527,9 +519,10 @@ def _closest_linked_lp_in_kd_tree_batched(
         points, linked_lps, tree: KDTree, k: int = 1
     ):
         p2ds = np.array([vec_2d(p) for p in points])
-        closest_indices = tree.query(
-            p2ds, k=min(k, len(linked_lps)), return_distance=False, sort_results=True
+        _, closest_indices = tree.query(
```

Comment:
Removing `scipy` entirely may be difficult. https://github.com/stefankoegl/kdtree This implementation can only find the nearest node instead of nearest k